src/sagemaker/clarify.py

```diff
@@ -393,6 +431,94 @@ def to_dict(self) -> Dict[str, Any]:  # pragma: no cover
         return segment_config_dict
 
 
+class TimeSeriesDataConfig:
+    """Config object for TimeSeries explainability specific data fields."""
+
+    def __init__(
+        self,
+        target_time_series: Union[str, int],
+        item_id: Union[str, int],
+        timestamp: Union[str, int],
+        related_time_series: Optional[List[Union[str, int]]] = None,
+        item_metadata: Optional[List[Union[str, int]]] = None,
+    ):
+        """Initialises TimeSeries explainability data configuration fields.
+
+        Args: #TODO: verify param descriptions are accurate
+            target_time_series (str or int): A string or a zero-based integer index.
+                Used to locate target time series in the shared input dataset.
+            item_id (str or int): A string or a zero-based integer index. Used to
+                locate item id in the shared input dataset.
+            timestamp (str or int): A string or a zero-based integer index. Used to
+                locate timestamp in the shared input dataset.
+            related_time_series (list[str] or list[int]): Optional. An array of strings
+                or array of zero-based integer indices. Used to locate all related time
+                series in the shared input dataset.
+            item_metadata (list[str] or list[int]): Optional.  An array of strings or
+                array of zero-based integer indices. Used to locate all item metadata
+                fields in the shared input dataset.
```

Comment:
For the timebeing we use `str` for jmespaths, right?

For `List[int]` should you add a check there is no overlap, let's say, between tts and rts?

Reply:
Also, please fix this here (and in the container's package)
```
SchemaOptional("related_time_series"): [Or(int, str)],
SchemaOptional("item_metadata"): [Or(int, str)],
```

Also, let's add Unit Tests to validate config can be created correctly.

Reply:
> Also, please fix this here (and in the container's package)
> 
> ```
> SchemaOptional("related_time_series"): [Or(int, str)],
> SchemaOptional("item_metadata"): [Or(int, str)],
> ```

To confirm, `related_time_series` and `item_metadata` should no longer be lists?

> Also, let's add Unit Tests to validate config can be created correctly.

I believe I already have unit tests for `TimeSeriesDataConfig` creation, see lines 343-500 of `test_clarify.py`. Unless that's not what you meant?

Reply:
1. they should be lists, like you defined in your Python classes
2. you have `list[int]` in python objects, but `int` in this schema. why did not the test capture this?

Reply:
I'm pretty sure the square brackets around the schema entry means it is a list.
For example: 
```
SchemaOptional("item_metadata"): [Or(int, str)],
```
refers to a list but
```
SchemaOptional("group_variable"): Or(str, int),
```
refers to a single item. Unless I'm wrong?